credsweeper/scanner/scan_type/scan_type.py

```diff
@@ -81,32 +79,17 @@ def get_line_data(
             LineData object if pattern a line and filters do not remove current line. None otherwise
 
         """
-        if not cls.is_pattern_detected_line(target.line, pattern):
-            return None
-        logger.debug("Valid line for pattern: %s in file: %s:%d in line: %s", pattern, target.file_path,
-                     target.line_num, target.line)
-        line_data = LineData(config, target.line, target.line_pos, target.line_num, target.file_path, target.file_type,
-                             target.info, pattern)
-
-        if cls.filtering(config, target, line_data, filters):
-            return None
-        return line_data
-
-    @classmethod
-    def is_pattern_detected_line(cls, line: str, pattern: re.Pattern) -> bool:
-        """Check if pattern present in the line.
-
-        Args:
-            line: Line to check
-            pattern: Compiled regex object
-
-        Return:
-            Boolean. True if pattern is present. False otherwise
-
-        """
-        if pattern.search(line):
-            return True
-        return False
+        for _match in pattern.finditer(target.line):
+            logger.debug("Valid line for pattern: %s in file: %s:%d in line: %s", pattern, target.file_path,
+                         target.line_num, target.line)
+            line_data = LineData(config, target.line, target.line_pos, target.line_num, target.file_path,
+                                 target.file_type, target.info, pattern, _match)
```

Comment:
Adding `_match` to parameter of `LineData` reduces the `pattern.search()` calling count in half compared to the previous one! :eyes: